src/components/encoding-pane/range-filter-shelf.tsx

```diff
@@ -1,46 +1,123 @@
+import {ExpandedType} from 'compassql/build/src/query/expandedtype';
+import * as Slider from 'rc-slider';
 import * as React from 'react';
+import * as CSSModules from 'react-css-modules';
 import {DateTime} from 'vega-lite/build/src/datetime';
 import {RangeFilter} from 'vega-lite/build/src/filter';
-import {FILTER_MODIFY_MAX_BOUND, FILTER_MODIFY_MIN_BOUND, FilterAction} from '../../actions/filter';
+import {FILTER_MODIFY_BOTH_BOUNDS, FILTER_MODIFY_MAX_BOUND, FILTER_MODIFY_MIN_BOUND,
+  FilterAction} from '../../actions/filter';
+import * as styles from './range-filter-shelf.scss';
 
 export interface RangeFilterShelfProps {
   domain: number[] | DateTime[];
   filter: RangeFilter;
   index: number;
+  type: ExpandedType;
   handleAction: (action: FilterAction) => void;
 }
 
 class RangeFilterShelfBase extends React.Component<RangeFilterShelfProps, {}> {
   public render() {
-    const {domain} = this.props;
+    const {domain, filter, type} = this.props;
+    const range = filter.range;
+    const lowerBound = Math.floor(domain[0] as number);
+    const upperBound = Math.ceil(domain[1] as number);
+    let currMin, currMax;
+    if (type === ExpandedType.TEMPORAL) {
+      currMin = new Date(range[0]);
+      currMax = new Date(range[1]);
+    } else {
+      currMin = range[0];
+      currMax = range[1];
+    }
+    const createSliderWithTooltip = Slider.createSliderWithTooltip;
+    const Range = createSliderWithTooltip(Slider.Range);
+
+    function formatTime(value: number) {
+      if (type === ExpandedType.TEMPORAL) {
+        return new Date(value).toString();
+      }
+      return value;
+    }
+
     return (
-      <div>
-        {domain}
+      <div styleName='range-filter-pane'>
+        <div>
+          <div styleName='bound'>
+            min: <a onClick={this.focusInput.bind(this, `${filter.field}min`)}><i className="fa fa-pencil"/></a>
```

Comment:
We normally do `.bind(this)` in constructor, which mean binding only happen once.  
Doing it in event handler like this will cause .bind() to run every single time. 

Same for the rest of this file.